TombEngine/Game/Lara/lara_fire.cpp

```diff
@@ -1049,28 +1049,30 @@ void LaraTargetInfo(ItemInfo* laraItem, const WeaponInfo& weaponInfo)
 
 void HitTarget(ItemInfo* laraItem, ItemInfo* targetEntity, GameVector* hitPos, int damage, bool isExplosive)
 {
-	const auto& lara = *GetLaraInfo(laraItem);
 	const auto& object = Objects[targetEntity->ObjectNumber];
 
 	targetEntity->HitStatus = true;
-
 	if (targetEntity->IsCreature())
 		GetCreatureInfo(targetEntity)->HurtByLara = true;
 
 	if (hitPos != nullptr)
 	{
-		int foundJointID = -1;
-		for (int jointID = 0; jointID < object.nmeshes; jointID++)
+		hitPos->RoomNumber = targetEntity->RoomNumber;
+
+		int foundJointIndex = -1;
+		for (int jointIndex = 0; jointIndex < object.nmeshes; jointIndex++)
 		{
-			auto pos = GetJointPosition(targetEntity, jointID);
-			if (Vector3i::Distance(hitPos->ToVector3i(), pos) < BLOCK(1 / 16.0f))
+			auto pos = GetJointPosition(targetEntity, jointIndex);
+			float maxDist = targetEntity->ItemFlags[7] != 0 ? (float)targetEntity->ItemFlags[7] : BLOCK(1 / 16.0f);
```

Comment:
What is this? What is `ItemFlags[7]`? This looks like a hack.

Reply:
fixed